content_purge.vcl

```diff
@@ -0,0 +1,101 @@
+        }
+    }
+}
+
+sub handle_simple_purge_requests {
+    # Ban the provided URL as is.
+    ban("req.http.host == " + req.http.host + " && req.url == " + req.url);
+    # The provided URL usually has no trailing slash; also invalidate variants having one.
+    if (req.url !~ "/+$") {
+        ban("req.http.host == " + req.http.host + " && req.url == " + req.url + "/");
+    }
+    std.log(req.http.host + req.url + "PURGED");
+        return (synth(200, "PURGED"));
+}
+
+# Handle PURGE requests emitted by the "Flexible Purge" Drupal module.
+sub handle_flexible_purge_requests {
+    call check_invalidate_headers;
+    std.log("Purge accepted flexible");
+    if (req.http.X-Invalidate-Type == "full") {
+        if (req.http.X-Invalidate-Tag) {
+            std.log("Tag based purge");
+            ban("req.http.X-Application-Tag == " + req.http.X-Invalidate-Tag);
+            std.log(req.http.X-Invalidate-Tag + "PURGED");
+                return (synth(200, "PURGED"));
+        }
+        elseif (req.http.X-Invalidate-Host && req.http.X-Invalidate-Base-Path) {
+            std.log(req.http.X-Invalidate-Host + req.http.X-Invalidate-Base-Path + "PURGED");
+            ban("req.http.host == " + req.http.X-Invalidate-Host + " && req.url ~ ^" + req.http.X-Invalidate-Base-Path);
+                return (synth(200, "PURGED"));
+        }
+    }
+    elseif (req.http.X-Invalidate-Type ~ "^(wildcard|regexp-(multiple|single))$") {
+        if (req.http.X-Invalidate-Regexp) {
+            if (req.http.X-Invalidate-Tag) {
+                std.log("X-Invalidate-Tag regex based purge");
+                ban("req.http.X-Application-Tag == " + req.http.X-Invalidate-Tag + " && req.http.X-FPFIS-Drupal-Path ~ " + req.http.X-Invalidate-Regexp);
+                std.log(req.http.X-Invalidate-Tag + req.http.X-Invalidate-Regexp + "PURGED");
+                    return (synth(200, "PURGED"));
+            }
+            else if (req.http.X-Invalidate-Host) {
+                std.log("X-Invalidate-xxx regex based purge");
+                ban("req.http.host == " + req.http.X-Invalidate-Host + " && req.url ~ " + req.http.X-Invalidate-Regexp);
+                std.log(req.http.X-Invalidate-Host + req.http.X-Invalidate-Regexp + "PURGED");
+                    return (synth(200, "PURGED"));
+            }
+        }
+    }
+    else {
+        std.log("ERROR, bad type:" + req.http.X-Invalidate-Type);
+        return (synth(400, "ERROR"));
+    }
+}
+
+# Sanitize known headers with the intent of using them to compose ban
+# statements.
+# Ban statements generated from VCL code do not need to have their arguments
+# surrounded with double quotes (this makes sense only when working with CLI
+# tools such as varnishadm).
+# Still, we must prevent ban injections by refusing to include HTTP headers
+# containing " && " into the ban statement as they are very likely to be an
+# injection attempt.
+sub check_invalidate_headers {
+    if (
```

Comment:
Merge in 1 condition (just to grep 405 in conf)
Debug added for each treatment on this file